cylc/flow/scripts/config.py

```diff
@@ -114,13 +134,50 @@ def get_config_file_hierarchy(reg: Optional[str] = None) -> List[str]:
     return filepaths
 
 
+def options_are_valid(
+    options: 'Values', reg: Optional[str] = None
+) -> bool:
+    """Check parser for options that should not be allowed.
+    """
+    from cylc.flow.exceptions import UserInputError
+    from cylc.flow import LOG
+    if (
+        options.print_platform_names and
+        options.print_platform_meta
+    ):
+        LOG.error(UserInputError(
+            '--platform-names is not compatible with --platform-meta'
+        ))
+        return False
```

Comment:
Raise the exception rather than logging it.

```suggestion
      raise (UserInputError(
            '--platform-names is not compatible with --platform-meta'
        )
```